File: group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map <string, vector<string>> umap;
        vector <vector<string>> ans;
        for (string& w:strs) {
            string tmp = w;
            sort(tmp.begin(), tmp.end());
            umap[tmp].push_back(w);
        }
        ans.reserve(umap.size());
        for (auto& item:umap) {
            ans.push_back(move(item.second));
        }
        return ans;
        
    }
};

File: group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}
        
        for s in strs:
            chars = sorted(s)
            key = "".join(chars)
            if key not in dict:
                dict[key] = []
            dict[key].append(s)
        
        
        return (list(dict.values()))